recipes/full_finetune_single_device.py

```diff
@@ -197,7 +197,7 @@ def load_checkpoint(self, cfg_checkpointer: DictConfig) -> Dict[str, Any]:
         """
         self._checkpointer = config.instantiate(
             cfg_checkpointer,
-            resume_from_checkpoint=self._resume_from_checkpoint,
+            should_load_recipe_state=self._resume_from_checkpoint,
```

Comment:
this will break every config or system that may be using this function. Although i agree its a better name, i dont think it is worth to change it in this PR. A possibly better alternative is to keep both, handle it in the class, and raise a deprecation warning. We should check what others think.

Reply:
If users have custom recipes with the resume_from_checkpoint, then it may be better to deprecate the resume_from_checkpoint and introduce should_load_recipe_state, otherwise it will be a backward incompatible change. However if thats not the case, then it seems okay to me to update all of our recipes which is a low risk since there is no functionality change. 

cc @ebsmothers who had this comment in the earlier version of the PR.